beatmap_reader/hitobject/std/std_holdnote_hitobject_base.py

```diff
@@ -87,29 +133,55 @@ def __process_slider_data(self, sdata):
         return curve_type, curve_points
 
     
-    def __process_curve_points(self, curve_type, curve_points, px_len):
-        gen_points  = []
-        
+    @staticmethod
+    def __process_curve_points(curve_type, curve_points, px_len):
         if curve_type == StdHoldNoteHitobjectBase.BEZIER:
-            return self.__make_bezier(curve_points)
+            return StdHoldNoteHitobjectBase.__make_bezier(curve_points)
 
         if curve_type == StdHoldNoteHitobjectBase.CIRCUMSCRIBED:
             if len(curve_points) == 3:
-                gen_points = self.__make_circumscribed(curve_points, px_len)
-                if len(gen_points) == 0:
-                    return self.__make_bezier(curve_points)
-                return gen_points
-            
-            return self.__make_bezier(curve_points)
+                return StdHoldNoteHitobjectBase.__make_circumscribed(curve_points, px_len)
+            return StdHoldNoteHitobjectBase.__make_bezier(curve_points)
 
         if curve_type == StdHoldNoteHitobjectBase.LINEAR1:
-            return self.__make_linear(curve_points)
+            return StdHoldNoteHitobjectBase.__make_linear(curve_points)
 
         if curve_type == StdHoldNoteHitobjectBase.LINEAR2:
-            return self.__make_linear(curve_points)
-
-
-    def __make_linear(self, curve_points):
+            print('WARN[beatmap_reader]: found catmull, treating as linear')
+            return StdHoldNoteHitobjectBase.__make_linear(curve_points)
+
+
+    @staticmethod
+    def __calculate_length_sums(gen_points):
+        length_sums = [ 0 ]
+        for i in range(len(gen_points) - 1):
+            distance = dist(gen_points[i], gen_points[i + 1])
+            length_sums.append(length_sums[-1] + distance)
+        return length_sums
+
+
+    @staticmethod
+    def __snap_path_length(gen_points, length_sums, px_len, extend):
```

Comment:
Should be renamed to some like  `__process_snap_path_length`. Needs something to indicate that `gen_points` and `length_sums` are being modified/generated by this function.

Reply:
Does `__process_curve_length` seem reasonable? Also for indicating the argument mutation is a doc-comment explaining what it does enough?

Reply:
Yes, and yes